pkg/ccl/backupccl/backuputils/memory_backed_quota_pool_test.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Licensed as a CockroachDB Enterprise file under the Cockroach Community
+// License (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//     https://github.com/cockroachdb/cockroach/blob/master/licenses/CCL.txt
+
+package backuputils
+
+import (
+	"context"
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/settings/cluster"
+	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
+	"github.com/cockroachdb/cockroach/pkg/util/mon"
+	"github.com/cockroachdb/cockroach/pkg/util/quotapool"
+	"github.com/cockroachdb/cockroach/pkg/util/syncutil"
+	"github.com/cockroachdb/errors"
+)
+
+// TestMemoryBackedQuotaPoolBasic tests the basic functionality
+// of the MemoryBackedQuotaPool in the presence of multiple
+// goroutines acquiring and releasing quota, also with multiple
+// goroutines acquiring and releasing memory from the parent
+// memory monitor.
+func TestMemoryBackedQuotaPoolBasic(t *testing.T) {
```

Comment:
this is a fancy test, I think you need a simple non-concurrent deterministic api unit tests, so that you can say allocate X and then verify it succeeded or failed, and of course verify more advanced scenarios.

a test like here may exercise more code path but also, it might be flaky and it will be hard to say what the issue is when it fails. also, nothing guarantees that the execution order of those goroutines will create a meaningful test - instead we may first run all the goroutines that grow and shrink the pool and then those that ask for quota. I understand it's not likely with 100 goroutines but overall my point is that a deterministic test can assert on the state of the object better. WDYT? did I miss anything about this test?